src/hud_element.c

```diff
@@ -229,15 +798,970 @@ void update_hud_elements(void) {
+            hudElement->paletteAddr = (u8*)*nextPos++;
+            hudElement->readPos = (HudScript*)nextPos;
+
+            if (hudElement->flags & HUD_ELEMENT_FLAGS_MEMOFFSET) {
+                hudElement->imageAddr += hudElement->memOffset;
+                hudElement->paletteAddr += hudElement->memOffset;
+            }
+
+            if (hudElement->flags & HUD_ELEMENT_FLAGS_FIXEDSCALE) {
+                if ((hudElement->flags & HUD_ELEMENT_FLAGS_CUSTOM_SIZE) == 0) {
+                    tileSizePreset = hudElement->tileSizePreset;
+                    drawSizePreset = hudElement->drawSizePreset;
+                    imageWidth = gHudElementSizes[tileSizePreset].width;
+                    imageHeight = gHudElementSizes[tileSizePreset].height;
+                    drawWidth = gHudElementSizes[drawSizePreset].width;
+                    drawHeight = gHudElementSizes[drawSizePreset].height;
+                } else {
+                    imageWidth = hudElement->customImageSize.x;
+                    imageHeight = hudElement->customImageSize.y;
+                    drawWidth = hudElement->customDrawSize.x;
+                    drawHeight = hudElement->customDrawSize.y;
+                }
+
+                if ((hudElement->flags & HUD_ELEMENT_FLAGS_200) == 0) {
+                    hudElement->flags |= HUD_ELEMENT_FLAGS_200;
+                    hudElement->unkImgScale[0] = imageWidth;
+                    hudElement->unkImgScale[1] = imageHeight;
+                    hudElement->deltaSizeX = ((f32)drawWidth - (f32)imageWidth) / (f32)hudElement->updateTimer;
+                    hudElement->deltaSizeY = ((f32)drawHeight - (f32)imageHeight) / (f32)hudElement->updateTimer;
+                } else {
+                    hudElement->flags &= ~HUD_ELEMENT_FLAGS_200;
+                    hudElement->unkImgScale[0] = drawWidth;
+                    hudElement->unkImgScale[1] = drawHeight;
+                    hudElement->deltaSizeX = ((f32)imageWidth - (f32)drawWidth) / (f32)hudElement->updateTimer;
+                    hudElement->deltaSizeY = ((f32)imageHeight - (f32)drawHeight) / (f32)hudElement->updateTimer;
+                }
+            }
+            break;
+
+        case HUD_ELEMENT_OP_SetImage:
+            hudElement->updateTimer = *nextPos++;
 
-INCLUDE_ASM(void, "hud_element", render_hud_elements_backUI, void);
+            if ((hudElement->flags & HUD_ELEMENT_FLAGS_BATTLE) == 0) {
+                entryRaster = gHudElementCacheTableRasterWorld;
+                entryPalette = gHudElementCacheTablePaletteWorld;
+            } else {
+                entryRaster = gHudElementCacheTableRasterBattle;
+                entryPalette = gHudElementCacheTablePaletteBattle;
+            }
 
-INCLUDE_ASM(void, "hud_element", render_hud_elements_frontUI, void);
+            i = 0;
+            while (TRUE) {
+                if (entryRaster[i].id == *nextPos) {
+                    break;
+                }
+                ASSERT(++i < 192);
```

Comment:
I'm seeing 192 a lot. If we can't do ARRAY_COUNT, it might be good to have it as a define